src/main/java/com/devonfw/ide/sonarqube/common/impl/check/thirdparty/DevonArchitecture3rdPartyHibernateCheck.java

```diff
@@ -69,9 +65,14 @@ private boolean isUsingProprietaryHibernateAnnotation(String targetSimpleName, S
   private boolean isNotImplementingHibernateEnversInImplScope(JavaType source, String targetSimpleName,
       String targetPackageName) {
 
-    return !source.isScopeImpl()
+    return (targetPackageName.startsWith("org.hibernate.envers")) && !source.isScopeImpl()
         && (!targetPackageName.equals("org.hibernate.envers") || targetSimpleName.startsWith("Default")
             || targetSimpleName.contains("Listener") || targetSimpleName.contains("Reader"));
   }
 
+  private boolean isImplementingHibernateEnversInternalsDirectly(String targetPackageName) {
+
+    return targetPackageName.startsWith("org.hibernate.envers") && targetPackageName.contains("internal");
+  }
+
```

Comment:
Here I do not really see that quality was improved. Checks like `targetPackageName.startsWith("org.hibernate.envers")` are now duplicated.
What was SonarQube complaining about here so we did this change?

Reply:
True, I will rework the methods so that they don't check double for something. The problem here is that the complexity of the checkDependency() method is too high. So my thinking was to put some of the if-statements into seperate methods, so that checkDependency() is not full of if-statements.

Reply:
The problem is that if I put the if-check for 'envers' at the top and nest the other two checks into this if-branch (the way it was before), the complexity increases to 16 from the 15 allowed.

Reply:
Cyclomatic-Complexity-Rules are actually a pain. If they drive developers to make changes that are not motivated by improving the quality but shutting up a stupid rule...
However, lets not make this too complicated. Your PR is fine and I can now approve.